shared/test_run_filter_test.go

```diff
@@ -0,0 +1,54 @@
+// +build small
+
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTestRunFilter_NextPage_MaxCount(t *testing.T) {
+	ten := 10
+	filter := TestRunFilter{
+		MaxCount: &ten,
+	}
+	chrome, _ := ParseProductSpec("chrome")
+	loadedRuns := TestRunsByProduct{
+		ProductTestRuns{
+			Product:  chrome,
+			TestRuns: make(TestRuns, 10),
+		},
+	}
+	assert.Equal(t, &TestRunFilter{
+		MaxCount: &ten,
+		Offset:   &ten,
+	}, filter.NextPage(loadedRuns))
+}
+
+func TestTestRunFilter_NextPage_From(t *testing.T) {
```

Comment:
Can you also add a test for paginating aligned runs where the range of times of the individual runs in each aligned group overlap? I expect that will not work without some extra work, because some of the runs that will be needed to discover the next aligned run are excluded by the filtering. Given how slow (>24 h) Edge runs have sometimes been I expect this has happened and that some aligned runs would be missed in pagination because of it.

Reply:
I think the simplest solution here is to include the run in the time-range page that does include the first run for the SHA. I'll add a test and implement that.